src/orchard/resource.clj

```diff
@@ -0,0 +1,73 @@
+(ns orchard.resource
+  "Resolve JVM resource-related information."
+  {:added "0.5.0"}
+  (:require
+   [clojure.java.io :as io]
+   [orchard.classpath :as cp]))
+
+(defn- trim-leading-separator
+  "Trim the java.io.File/separator at the beginning of s."
+  [s]
+  (if (.startsWith s java.io.File/separator)
+    (subs s 1)
+    s))
+
+(defn- project-resources
+  "Get a list of classpath resources."
+  []
+  (mapcat
+   (fn [directory]
+     (->> directory
+          (file-seq)
+          (filter (memfn isFile))
+          (map (fn [file]
+                 (let [relpath (-> file
+                                   (.getPath)
+                                   (.replaceFirst
+                                    (.getPath directory)
+                                    "")
+                                   (trim-leading-separator))]
+                   {:root directory
+                    :file file
+                    :relpath relpath
+                    :url (io/resource relpath)})))
+          (remove #(.startsWith (:relpath %) "META-INF/"))
+          (remove #(re-matches #".*\.(clj[cs]?|java|class)" (:relpath %)))))
+   (filter (memfn isDirectory) (map io/as-file (cp/classpath (cp/boot-aware-classloader))))))
+
+(defn resource-full-path [relative-path]
+  (io/resource relative-path (cp/boot-aware-classloader)))
+
+(defn resource-path-tuple
+  "If it's a resource, return a tuple of the relative path and the full
+  resource path."
+  [x]
+  (or (if-let [full (resource-full-path x)]
+        [x full])
+      (if-let [[_ relative] (re-find #".*jar!/(.*)" x)]
+        (if-let [full (resource-full-path relative)]
+          [relative full]))
+      ;; handles load-file on jar resources from a cider buffer
+      (if-let [[_ relative] (re-find #".*jar:(.*)" x)]
+        (if-let [full (resource-full-path relative)]
+          [relative full]))))
+
+(defn resource-path
+  "Return the resource path for the given input map.
+
+  It requires the :name key."
+  {:added "0.5.0"}
+  [{:keys [name]}]
```

Comment:
Why does this take a map?

Reply:
this map was originally the message, but given the new take on things I will just use the name